README.md

```diff
@@ -197,15 +252,17 @@ const useStore = create<StoreState>(
 
 #### **Prevent unchanged states to be stored**
 
-`equality?: (pastState: TState, currentState: TState) => boolean`
+`equality?: (pastState: PartialTState, currentState: PartialTState) => boolean`
 
-For performance reasons, you may want to use a custom `equality` function to determine when a state change should be tracked. You can write your own or use something like [`fast-equals`](https://github.com/planttheidea/fast-equals), [`fast-deep-equal`](https://github.com/epoberezkin/fast-deep-equal), [`zustand/shallow`](https://github.com/pmndrs/zustand/blob/main/src/shallow.ts), [`lodash.isequal`](https://www.npmjs.com/package/lodash.isequal), or [`underscore.isEqual`](https://github.com/jashkenas/underscore/blob/master/modules/isEqual.js). By default, all state changes to your store are tracked.
+Provide an `equality` function to determine when a state change should be tracked.
```

Comment:
I removed "performance reasons", as I think many reasons to use a custom equality function are not actually for performance reasons. I think many instances of using an equality function are to ensure that effectively duplicate state histories are not tracked. For example, by using deep-equality fns.

Reply:
Thanks for the explanation! Could you find a way to include possible reasons someone might use this option? I included `performance` originally because people asked for that reason and I wanted people to Cmd+F `performance` in the readme. However, I agree there are other important reasons too, so could you include more of them? Thank you!

Reply:
No problem! I expanded the section a little bit, and added back the word `performance` for easy Cmd+F-ing -- let me know fi you think it is clear or if I should try to expand it more. I'm trying to find the right balance of not bloating the code examples while also trying to be as immediately clear to the reader as possible. If you think it would be even clearer to provide example `StoreState` interfaces for the various use-cases, I'm happy to add those. But if it seems to be sufficient without them, I think my instinct is to try to stay as lean as possible.

Reply:
Thanks for making that change! Striking the balance of keeping docs clear but bloated is difficult and I appreciate you for trying to find that balance too.